cms/server/admin/handlers/trainingprogram.py

```
@@ -100,6 +100,23 @@ def post(self, training_program_id: str):
+
+            if start_str:
+                from datetime import datetime as dt
+                training_program.managing_contest.start = dt.strptime(start_str, "%Y-%m-%dT%H:%M")
+
+            if stop_str:
+                from datetime import datetime as dt
+                training_program.managing_contest.stop = dt.strptime(stop_str, "%Y-%m-%dT%H:%M")
+
+            # Validate that stop is not before start
+            if training_program.managing_contest.stop < training_program.managing_contest.start:
+                raise ValueError("End time must be after start time")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)